BCStudentSoftwareDevTeam/cas
# 432 fix division scrolling

# Issue


|  |  |
|:------------------------------------------|-----------------|
| In development, the sidebar for the courses page has a few formatting issues and its behavior is centered around the structure of the data rather than the use case. This pull request addresses both the structural flaws and the use case problems. <br> <br> ### Structural Issues <br> Notice that the sidebar's scrollbar doesn't extend all the way to the bottom of the page. This led to some confusing navigation where you could be at the bottom of the page but not see the bottom of the sidebar. What we did: <br> <br> ![image](https://github.com/user-attachments/assets/e8513983-76a9-4adb-8935-6f65dedaf65c) <br> <br> ### Use Case Issues <br> The user does not care about what division their course abbreviation is in and these menus just make it harder to find the course they're searching for <br> <br> <br> | Old state of the sidebar: ![image](https://github.com/user-attachments/assets/42caad84-6c67-40bc-b616-f1549b3d595a) |


# Solution
### Structural Issues
We restructured the base page so that the sidebar div actually fills up the whole height of the screen.

![image](https://github.com/user-attachments/assets/d48cf586-e199-4ebb-b6b9-5e43cfd73c15)

### Use Case Issues
We created a select picker which lets you click the header and then simply search for the course abbreviation you want to look at. This field is searchable by the course abbreviation, division name, and division number.

![image](https://github.com/user-attachments/assets/43f74214-d95a-496e-aa14-26b6f5a61b95)

# To Test

1. Checkout our branch. ```git checkout 432-Fix-Division-Scrolling```
2. Pull our branch. ```git pull origin 432-Fix-Division-Scrolling```
3. Navigate to the courses page using the navbar.
![image](https://github.com/user-attachments/assets/f42db2aa-29e4-4995-b6bb-b0518f5a50fa)
4. Select any term when prompted.
5. Ensure that we can search for any course abbreviation on the sidebar and make sure the formatting looks good and feels intuitive.

Resolves Issue #432
